azure-core/README.md

```diff
@@ -0,0 +1,357 @@
+
+
+# Azure Core Library
+
+## Pipeline
+
+The Azure Core pipeline is a re-strucuting of the msrest pipeline introduced in msrest 0.6.0.
+Further discussions on the msrest implementation can be found in the [msrest wiki](https://github.com/Azure/msrest-for-python/wiki/msrest-0.6.0---Pipeline).
+
+The Azure Core Pipeline is an implementation of chained policies as described in the [Azure SDK guidelines](https://github.com/Azure/azure-sdk/tree/master/docs/design).
+
+The Python implementation of the pipeline has some mechanisms specific to Python. This is due to the fact that both synchronous and asynchronous implementations of the pipeline must be supported indepedently.
+
+When constructing an SDK, a developer may consume the pipeline like so:
+
+```python
+from azure.core import Configuration, Pipeline
+from azure.core.transport import RequestsTransport, HttpRequest
+from azure.core.pipeline.policies import (
+    UserAgentPolicy,
+    HeadersPolicy,
+    RetryPolicy,
+    RedirectPolicy,
+    ContentDecodePolicy
+)
+
+class FooServiceClient():
+
+    @staticmethod
+    def create_config(**kwargs):
+        # Here the SDK developer would define the default
+        # config to interact with the service
+        config = Configuration(**kwargs)
+        config.user_agent_policy = UserAgentPolicy("ServiceUserAgentValue", **kwargs)
+        config.headers_policy = HeadersPolicy({"CustomHeader": "Value"})
+        config.retry_policy = RetryPolicy(**kwargs)
+        config.redirect_policy = RedirectPolicy(**kwargs)
+        config.logging_policy = NetworkTraceLoggingPolicy(**kwargs)
+        config.proxy_policy = ProxyPolicy(**kwargs)
+
+    def __init__(self, endpoint=None, credentials=None, configuration=None, transport=None, **kwargs):
+        config = configuration or FooServiceClient.create_config(**kwargs)
+        transport = RequestsTransport(config)
+        policies = [
+            config.user_agent_policy,
+            config.headers_policy,
+            credentials,
+            ContentDecodePolicy(),
+            config.redirect_policy,
+            config.retry_policy,
+            config.logging_policy,
+        ]
+        self._pipeline = Pipeline(transport, policies=policies)
+
+    def get_request(self, **kwargs)
```

Comment:
I would suggest that we rename get_request to something else/slightly less generic. As written, it is not clear if this is part of the infrastructure or if it is a service specific method.